### ISSUE TITLE
Provide an option that specifies the redirect waiting time

### ISSUE THREAD

**Issue by amits1995** [DevExpress/testcafe#2508]
### Are you requesting a feature or reporting a bug?
reporting a bug

### What is the current behavior?
I'm taking two screenshots (pre-login and post-login)
The only screenshot being save is the pre-login and the content of the image is of post-login screen

### What is the expected behavior?
having two screenshots. one of the login page and one of the app main page

### How would you reproduce the current behavior (if this is a bug)?
run the test

#### Provide the test code and the tested page URL (if applicable)

Tested page URL:
https://app.rookout.com/

Test code

```js
fixture`Rookout E2E Regression & Demo Flow`.page`https://app.rookout.com/`
test("sanity rookout", async t => {
    await t.takeScreenshot("pre-login")

    await login(t) // I obviously can't help you reproduce that

    await t.takeScreenshot("post-login")
})
```

### Specify your

* operating system: Windows 10
* testcafe version: 0.20.3
* node.js version: 8.11.1  
  
BTW the screenshot works perfectly on your site (https://testcafe.devexpress.com)  
It just breaks on our site

**Comment by LavrovArtem** [comment]
Hi @amits1995,
I've reproduced this issue and will investigate it. I think it can be related to a redirect from `https://app.rookout.com/` to `https://app.rookout.com/web_login`. You can use one of two workarounds:
1. Change the target page from ``.page`https://app.rookout.com/` `` to ``.page`https://app.rookout.com/web_login` ``
2. Place the `.navigateTo("/web_login")` step before `.takeScreenshot("pre-login")`

**Comment by amits1995** [comment]
Thank you very much!  
Workaround #1 seems to work

**Comment by AndreyBelym** [comment]
Hi @amits1995! I think I've found the reason of the problem: the redirect to the login page happens too late to be detected by TestCafe. I've increased the amount of time that TestCafe waits for a possible redirect request, so could you please try to install a debug build and report the feedback here? Just specify the following URL:

https://github.com/AndreyBelym/testcafe/releases/download/gh-2508/testcafe-0.20.2-1.tgz

as a version of TestCafe in your `package.json` file.

**Comment by miherlosev** [comment]
After discussing in private, we decided to convert this bug to an enhancement.

**Comment by amits1995** [comment]
Hey guys,
We have migrated our authentication logic to [Auth0](https://auth0.com) which means we redirect to a dedicated login page.  
This change seems to break the screenshots mechanism as `t.takeScreenshot("screenshot")` raises a null exception.  
I tried running our tests with the testcafe version you mentioned and it still doesn't work.  
Any idea?  
I'm working on a public site to help you guys reproduce my problem, will publish once it's done
